src/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -888,7 +889,11 @@ else if (list.size() == 0)
 		for (String key: list.get(0).keySet()) {
 			Object value = list.get(0).get(key);
 			if ( key.equals(languageIdColumn) ) {
-				host.setProperty(Contentlet.LANGUAGEID_KEY, value);

```

Comment:
Most of the Number's in Java extends from java.lang.Number and it has a method called: longValue
you may want to consider to ask if it is a number, so get the longValue. Just an idea